File: app/src/main/java/com/donald/musictheoryapp/Music/ScoreView/SClef.java
package com.donald.musictheoryapp.Music.ScoreView;

import android.graphics.Canvas;

import com.donald.musictheoryapp.Music.MusicXML.Clef;

class SClef extends ScoreStamp
{
	private char m_Glyph;
	private float m_VerticalOffsetOnStaff;

	protected SClef(ScoreView view)
	{
		super(view);
		m_Glyph = U.G_CLEF;
		m_VerticalOffsetOnStaff = 2 * staffStepHeight();
	}

	protected void setClef(Clef clef)
	{
		switch(clef.sign)
		{
			case Clef.Sign.F:
				m_Glyph = U.F_CLEF;
				m_VerticalOffsetOnStaff = (clef.line - 1) * staffStepHeight() * 2;
				break;
			case Clef.Sign.C:
				m_Glyph = U.C_CLEF;
				m_VerticalOffsetOnStaff = (clef.line - 1) * staffStepHeight() * 2;
				break;
			case Clef.Sign.G:
				m_Glyph = U.G_CLEF;
				m_VerticalOffsetOnStaff = (clef.line - 1) * staffStepHeight() * 2;
				break;
			default:
				throw new IllegalArgumentException(clef + " is not valid.");
		}
	}

	@Override
	protected void onDraw(Canvas canvas, float posX, float posY)
	{
		canvas.drawText(String.valueOf(m_Glyph),
			posX,
			posY + glyphSize() - m_VerticalOffsetOnStaff,
			glyphPaint()
		);
	}
}

File: app/src/main/java/com/donald/musictheoryapp/music/scoreview/SClef.kt
package com.donald.musictheoryapp.music.scoreview

import android.graphics.Canvas
import com.donald.musictheoryapp.music.MusicXML.Clef
import java.lang.IllegalArgumentException

internal class SClef(view: ScoreView) : ScoreStamp(view) {

    private var glyph = U.G_CLEF
    private var clefStaffPosition = 0
    override var relContentWidth: Float = 0F

    fun setClef(clef: Clef) {
        if (clef.printObject) {
            when (clef.sign) {
                Clef.Sign.F -> {
                    glyph = U.F_CLEF
                    clefStaffPosition = (clef.line - 1) * 2
                }
                Clef.Sign.C -> {
                    glyph = U.C_CLEF
                    clefStaffPosition = (clef.line - 1) * 2
                }
                Clef.Sign.G -> {
                    glyph = U.G_CLEF
                    clefStaffPosition = (clef.line - 1) * 2
                }
                else -> {
                    glyph = U.G_CLEF
                    clefStaffPosition = 0
                    throw IllegalArgumentException("$clef is not valid.")
                }
            }
            relContentWidth = REL_CLEF_WIDTH
        } else {
            relContentWidth = 0F
        }
    }

    override fun onDraw(canvas: Canvas, posX: Float, posY: Float) {
        val verticalOffOnStaff = -clefStaffPosition * staffStepHeight + staffHeight
        canvas.drawText(
            glyph.toString(),
            posX,
            posY + verticalOffOnStaff,
            glyphPaint
        )
    }

}